addons/account_voucher/migrations/9.0.1.0/post-migration.py

```diff
@@ -71,9 +71,21 @@ def create_payments_from_vouchers(env):
         SET payment_id = av.id
         FROM account_voucher_line avl
         JOIN account_voucher av ON av.id = avl.voucher_id
-        WHERE avl.move_line_id=aml.id
+        WHERE avl.move_line_id = aml.id
         AND av.voucher_type IN ('receipt', 'payment')
-        AND av.state in ('draft', 'posted')
+        AND av.state IN ('draft', 'posted')
+        AND avl.reconcile
```

Comment:
@jbeficent Is this not overly complicated? Should this not be enough to take partial reconciliations into account? 
```
 68     env.cr.execute(                                                            
 69         """\                                                                   
 70         UPDATE account_move_line aml                                           
 71         SET payment_id = av.id                                                 
 72         FROM account_voucher_line avl                                          
 73         JOIN account_voucher av ON av.id = avl.voucher_id                      
 74         WHERE avl.move_line_id = aml.id                                        
 75         AND av.voucher_type IN ('receipt', 'payment')                          
 76         AND av.state IN ('draft', 'posted')                                    
 77         AND ((NOT aml.reconcile_id IS NULL)                                    
 78             OR (NOT aml.reconcile_partial_id IS NULL))                         
 79         """                                                                    
 80     ) 
```